### DISCUSSION TITLE
One mono signal to identical left and right stereo signal

### DISCUSSION THREAD

**Discussion by dosas** [pschatzmann/arduino-audio-tools#1681] [Q&A] [answered]
I am transmitting a mono signal (channel=1) which I would like to output as an identical stereo signal on headphones.

I could not find anything in the [existing functionality](https://github.com/pschatzmann/arduino-audio-tools/wiki/Splitting-and-Merging-Audio), the closest thing I could find is [InputMerge](https://github.com/pschatzmann/arduino-audio-tools/blob/main/src/AudioTools/AudioStreams.h#L1261) but I could not get it to work the audio is broken. Am I missing something obvious?

```
// AudioInfo info(sampleRate, 1, bits); the info from the esp now stream
AudioInfo infoOut(sampleRate, 2, bits);

I2SStream out;
InputMerge<int16_t> imerge;
ESPNowStream now;

StreamCopy copier(out, imerge);

void setup(void) {
  auto espNowCfg = now.defaultConfig();
  now.begin(espNowCfg);

  auto i2sCfg = out.defaultConfig(TX_MODE);  
  i2sCfg.copyFrom(infoOut);
  
  imerge.add(now);
  imerge.add(now);
  imerge.begin(infoOut);

  out.begin(i2sCfg);
}

void loop() {
  copier.copy();
}


```

**Comment by pschatzmann** [comment, marked as answer]
It's documented in the chapter [Converting the Format](https://github.com/pschatzmann/arduino-audio-tools/wiki/Converting-the-Data-Format)

**Comment by dosas** [comment]
Thank you for the lightning fast answer and of course this awesome library. Works fine now with `FormatConverterStream`
